upower: 0.99.10 -> 0.99.11

dtzWill opened the pull request:
###### Motivation for this change

https://gitlab.freedesktop.org/upower/upower/-/tags/UPOWER_0_99_11

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

dtzWill committed:
upower: 0.99.10 -> 0.99.11

https://gitlab.freedesktop.org/upower/upower/-/tags/UPOWER_0_99_11

jtojnar commented:
Could you move the following to `nativeBuildInputs`:

    intltool
    libxslt
    docbook_xsl
    gobject-introspection

and replace `intltool` by `gettext` as suggested in the news?

worldofpeace committed:
upower: cleanup

* correct inputs
* drop dbus-glib
  Has not been needed for a very long
  time.
* intltool -> gettext
* add optional libimobiledevice
* propagate glib
  It's in Requires
* dev output
* drop useSystemd

worldofpeace committed:
nixos/upower: don't use activation script to create statedir

Systemd now handles this completely.

worldofpeace committed:
nixos/upower: drop custom unit

All of these changes ++ are in the upstream unit
Also drop glib in path of unit as I don't believe
this should be needed anymore.

worldofpeace committed:
nixos/upower: drop resumeCommands hack

Introduced 7 years ago, it's likely the issue has
been fixed. Even possible this could cause issues
with recent upower.

worldofpeace commented:
@dtzWill Did that and then some :+1:

jtojnar commented on nixos/modules/services/hardware/upower.nix:
```diff
@@ -49,55 +52,7 @@ in
-
-    system.activationScripts.upower =
-      ''
-        mkdir -m 0755 -p /var/lib/upower
```
How does systemd handle this? I do not see `tmpfiles.d` or ~~hmm, I thought only some directives like [`WorkingDirectory`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#WorkingDirectory=) were created but now I cannot find docs for even that.~~

worldofpeace replied:
https://gitlab.freedesktop.org/upower/upower/commit/e06bfc6a120750ed629291ddcec02f62f178fab4

jtojnar commented:
Given that we hard depend on libgudev, which is glib wrapper around udev, which is part of systemd, `useSystemd = false` does not make sense anymore.

worldofpeace commented:
> Given that we hard depend on libgudev, which is glib wrapper around udev, which is part of systemd, `useSystemd = false` does not make sense anymore.

It seems that polkit and pulseaudio have a `useSystemd` as well. Wonder if the situation is similar to this one.

dtzWill commented:
@worldofpeace :heart_eyes: thanks, this is great! \o/
